lib/ansiblelint/formatters/__init__.py

```diff
@@ -1,20 +1,40 @@
 """Output formatters."""
+import os
+
 try:
     from ansible import color
 except ImportError:
     from ansible.utils import color
-from ansiblelint.utils import normpath
 
 
-class Formatter(object):
+class BaseFormatter(object):
+    """Formats ansible lint output.
+
+    Base class for output formatters.
+
+    Args:
+        base_dir: reference directory against which display relative path.
+        relative_path (bool): whether to show path as relatvie or absolute
+    """
+
+    def __init__(self, base_dir, relative_path):
+        self._base_dir = base_dir if relative_path else None
+
+    def normpath(self, path):
+        if self._base_dir:
+            return os.path.relpath(path, start=self._base_dir)
+        return path
```

Comment:
Maybe we could have shorter lines
```suggestion
        if not self._base_dir:
            return path
        return os.path.relpath(path, start=self._base_dir)
```